firebase-firestore/src/main/java/com/google/firebase/firestore/core/FirestoreClient.java

```diff
@@ -171,22 +172,20 @@ public boolean isTerminated() {
   }
 
   /** Starts listening to a query. */
-  public QueryListener listen(
-      Query query, ListenOptions options, EventListener<ViewSnapshot> listener) {
+  public ListenerRegistration listen(
+      Query query,
+      ListenOptions options,
+      @Nullable Activity activity,
+      AsyncEventListener<ViewSnapshot> listener) {
     this.verifyNotTerminated();
     QueryListener queryListener = new QueryListener(query, options, listener);
     asyncQueue.enqueueAndForget(() -> eventManager.addQueryListener(queryListener));
-    return queryListener;
-  }
-
-  /** Stops listening to a query previously listened to. */
-  public void stopListening(QueryListener listener) {
-    // Checks for terminate but does not raise error, allowing it to be a no-op if client is already
-    // terminated.
-    if (this.isTerminated()) {
-      return;
-    }
-    asyncQueue.enqueueAndForget(() -> eventManager.removeQueryListener(listener));
+    return ActivityScope.bind(
+        activity,
+        () -> {
+          listener.mute();
+          asyncQueue.enqueueAndForget(() -> eventManager.removeQueryListener(queryListener));
```

Comment:
The `enqueueAndForget` already has `no-op` semantics after terminate.

Reply:
Please make this a comment in the file.